src/strtab.rs

```diff
@@ -60,44 +61,65 @@ impl<'a> Strtab<'a> {
                 overflow
             )));
         }
-        Ok(Strtab {
-            bytes: &bytes[offset..end],
-            delim: ctx::StrCtx::Delimiter(delim),
-        })
-    }
-    #[cfg(feature = "alloc")]
-    /// Converts the string table to a vector, with the original `delim` used to separate the strings
-    pub fn to_vec(&self) -> error::Result<Vec<&'a str>> {
-        let len = self.bytes.len();
-        let mut strings = Vec::with_capacity(len);
+        let mut result = Self::from_slice_unsafe(bytes, offset, len, delim);
         let mut i = 0;
         while i < len {
-            let string = self.get(i).unwrap()?;
-            i = i + string.len() + 1;
-            strings.push(string);
+            let string = get_str(i, result.bytes, result.delim)?;
+            result.strings.push((i, string));
+            i += string.len() + 1;
         }
-        Ok(strings)
+        Ok(result)
+    }
+    #[cfg(feature = "alloc")]
+    /// Parses a `Strtab` with `bytes` as the backing string table, using `delim` as the delimiter between entries.
+    pub fn new(bytes: &'a [u8], delim: u8) -> error::Result<Self> {
+        Self::parse(bytes, 0, bytes.len(), delim)
     }
-    /// Safely parses and gets a str reference from the backing bytes starting at byte `offset`.
+    #[cfg(feature = "alloc")]
+    /// Converts the string table to a vector of parsed strings.
+    pub fn to_vec(&self) -> Vec<&'a str> {
```

Comment:
unfortunately this is still a breaking change, since return signature changed. i'm 50/50 on fence whether it's worth it (can just Ok wrap at end and it's non-breaking, but maybe it's fine to break?)

Reply:
Ironically, I was on the fence of breaking it even further by including the starting offset of each entry in the result.

Reply:
Ok so I've thought about this, and I think we should return `Result` here, so that we don't have a breaking change, and if we switch to using the enum, it will work without changes; if we get that done i'm ok with merging this and making a minor release asap.

As a bonus, we can perhaps construct the vector on the fly if it's `empty` (i.e., it was constructed via the unsafe method), but that is probably a niche usecase.

Reply:
Both done, reverted to use the `Result` return type and implemented a fallback.